src/fondant/component_spec.py

```diff
@@ -343,31 +416,54 @@ def to_string(self) -> str:
     @property
     def input_arguments(self) -> t.Mapping[str, Argument]:
         """The input arguments of the component as an immutable mapping."""
-        return types.MappingProxyType(
-            {
-                info["name"]: Argument(
-                    name=info["name"],
-                    description=info["description"],
-                    type=info["type"],
-                    default=info["default"] if "default" in info else None,
+        args = {}
```

Comment:
A lof of repeated code between input and output arguments, can be one put in one function

Reply:
Agreed but this will probably be removed once we use the component spec as base for building the arguments and not the KubeflowComponentSpec.

https://github.com/ml6team/fondant/issues/393